core/base/composition.cpp

```diff
@@ -43,12 +43,13 @@ namespace {
 template <typename ValueType, typename OpIterator, typename VecIterator>
 inline void allocate_vectors(OpIterator begin, OpIterator end, VecIterator res)
 {
-    for (auto it = begin; it != end; ++it, ++res) {
+    for (auto it = begin; it != end; ++it) {
         if (*res != nullptr && (*res)->get_size()[0] == (*it)->get_size()[0]) {
             continue;
         }
         *res = matrix::Dense<ValueType>::create(
             (*it)->get_executor(), gko::dim<2>{(*it)->get_size()[0], 1});
+        ++res;
```

Comment:
Is this correct? If `*res != nullptr` (and the other condition is satisfied), then we never increase `res` and check the next one.

Reply:
Good catch, I now inverted the condition to keep the same code structure